django/template/__init__.py

```diff
@@ -40,37 +40,50 @@
 """
 
 # Multiple Template Engines
-
 from .engine import Engine
 from .utils import EngineHandler
 
 engines = EngineHandler()
 
-__all__ = ("Engine", "engines")
-
 
 # Django Template Language
 
-# Public exceptions
-from .base import VariableDoesNotExist  # NOQA isort:skip
-from .context import Context, ContextPopException, RequestContext  # NOQA isort:skip
-from .exceptions import TemplateDoesNotExist, TemplateSyntaxError  # NOQA isort:skip
+# Import the .autoreload module to trigger the registrations of signals.
+from . import autoreload  # noqa: E402
 
 # Template parts
-from .base import (  # NOQA isort:skip
+# Public exceptions
```

Comment:
The comments aren't in order anymore. I wasn't expecting the imports to be reordered as part of this ticket, but maybe it's okay.